observability/interceptor.go

```diff
@@ -49,11 +48,14 @@ func (i *Interceptor) WrapUnary(next connect.UnaryFunc) connect.UnaryFunc {
 			return nil, err
 		}
 
-		if verb := req.Header().Get(headers.VerbHeader); verb != "" {
-			metricsErr := i.recordVerbCallMetrics(ctx, verb, start, req.Header())
-			if metricsErr != nil {
-				logger.Errorf(metricsErr, "Failed to record metrics for verb: %s", verb)
-			}
+		callers, err := headers.GetCallers(req.Header())
+		if err != nil {
+			logger.Errorf(err, "Failed to get callers from headers")
```

Comment:
Why is this logged? If there's an invalid header, which is the only error that can occur, this should return an error.